src/Serializer/AbstractItemNormalizer.php

```diff
@@ -170,13 +170,12 @@ public function supportsDenormalization($data, $type, $format = null, array $con
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
-        $context['api_denormalize'] = true;
-        $context['resource_class'] = $class;
+        $context['input_class'] = $class;
         $inputClass = $this->getInputClass($class, $context);
 
-        if (null !== $inputClass && null !== $dataTransformer = $this->getDataTransformer($data, $class, $context)) {
+        if (null !== $inputClass && null !== $dataTransformer = $this->getDataTransformer($data, $context['resource_class'], $context)) {
             $data = $dataTransformer->transform(
-                parent::denormalize($data, $inputClass, $format, ['resource_class' => $inputClass] + $context),
+                parent::denormalize($data, $inputClass, $format, ['input' => ['class' => null]] + $context), // ensure we don't transformData for nested relations
```

Comment:
I see, can't we just remove the input metadata from the context here?

Reply:
y, that would work fine too I think.

Reply:
Done.